packages/eslint-plugin/lib/rules/react-require-autocomplete.js

```diff
@@ -0,0 +1,137 @@
+const {findDefinition} = require('../utilities');
+
+const autocompleteInputTypes = [
+  'color',
+  'date',
+  'datetime-local',
+  'email',
+  'month',
+  'number',
+  'password',
+  'range',
+  'search',
+  'tel',
+  'text',
+  'time',
+  'url',
+  'week',
+];
+const autocompleteElementNames = ['input', 'TextField'];
```

Comment:
If we have other components that some libraries may want the rule applied to, I _think_ we could make this a setting.

Reply:
Ha, I was curious about whether we had the `autocomplete-valid` rule and it seems like we apply it on the [<Select> component](https://github.com/Shopify/web-configs/blob/9852a937a6b51edc60215dcc34c1b943cb648fab/packages/eslint-plugin/lib/config/rules/jsx-a11y.js#L75-L80) too!